msu/hooks/entity/tactical/entity.nut

```diff
@@ -0,0 +1,24 @@
+::MSU.MH.hook("scripts/entity/tactical/entity", function(q) {
+	q.create = @(__original) function()
+	{
+		__original();
+		this.getFlags().set("MSU_UID", ::MSU.Serialization.isLoading() ? null : ::MSU.Utils.__generateUID());
```

Comment:
This should probably still be in a deliberate `setUID()` function to make sure (as much as possible) that we always generate a new UID if we set one.

Reply:
I don't think that's important because this `create` and `onDeserialize` below are the only two places where it's gonna be set. Nevertheless, I've pushed a commit implementing your suggestion - take a look at that. But I think we don't need to do this and should probably leave it as I did it first.